server/routerlicious/packages/routerlicious-base/src/alfred/runner.ts

```diff
@@ -140,28 +143,57 @@ export class AlfredRunner implements IRunner {
 			});
 		}
 
+		this.stopped = false;
+
 		return this.runningDeferred.promise;
 	}
 
 	// eslint-disable-next-line @typescript-eslint/promise-function-async
-	public stop(): Promise<void> {
-		// Close the underlying server and then resolve the runner once closed
-		this.server.close().then(
-			() => {
-				this.runningDeferred.resolve();
-			},
-			(error) => {
-				this.runningDeferred.reject(error);
-			},
-		);
-
-		return this.runningDeferred.promise;
+	public async stop(caller?: string, uncaughtException?: any): Promise<void> {
+		if (this.stopped) {
+			return;
+		}
+		this.stopped = true;
+
+		try {
+			// Close the underlying server and then resolve the runner once closed
+			await this.server.close();
+			if (caller === "uncaughtException") {
+				this.runningDeferred?.reject({
+					uncaughtException: serializeError(uncaughtException),
+				}); // reject the promise so that the runService exits the process with exit(1)
+			} else {
+				this.runningDeferred?.resolve();
+			}
+			this.runningDeferred = undefined;
+			if (!this.runnerMetric.isCompleted()) {
+				this.runnerMetric.success("Alfred runner stopped");
+			}
+		} catch (error) {
+			if (!this.runnerMetric.isCompleted()) {
+				this.runnerMetric.error("Alfred runner encountered an error during stop", error);
```

Comment:
use serializeError(error) in line 174

Reply:
that is not needed because lumberjack interface takes care of serializing it